airflow/sentry.py

```diff
@@ -81,13 +81,17 @@ def __init__(self):
             sentry_celery = CeleryIntegration()
             integrations.append(sentry_celery)
 
-        dsn = conf.get("sentry", "sentry_dsn")
+        dsn = None
+        sentry_config_opts = conf.getsection("sentry") or {}
```

Comment:
getsection method returns `dict[str,str]`.  This means that options that are not strings will not work properly.
https://github.com/getsentry/sentry-python/blob/464ca8dda09155fcc43dfbb6fa09cf00313bf5b8/sentry_sdk/consts.py#L37
The following options will works:
```
ca_certs
dist
dsn
environment
http_proxy
https_proxy
ignore_errors
in_app_exclude
in_app_include
release
server_name
request_bodies
```
The following options will NOT works:
```
attach_stacktrace
before_breadcrumb
before_send
debug
default_integrations
integrations
max_breadcrumbs
propagate_traces
sample_rate
send_default_pii
shutdown_timeout
transport
with_locals
```
This can cause very difficult to debug problems. It would be great if we supported more options or if an option is not supported, we would report a user-friendly error message.

Reply:
> getsection method returns dict[str,str]
It's not quite true. Here (https://github.com/apache/airflow/blob/master/airflow/configuration.py#L422) you can see that code is going through `_section` dict and trying to cast strings to proper type.
But fair enough it's not working for these props (list, dict and custom type is not supported by getsection):
```
integrations
in_app_include
in_app_exclude
ignore_errors
before_breadcrumb
before_send
transport
```
I will add some guard exception for these props.